pre_commit_hooks/check_twincat_versions.py

```diff
@@ -0,0 +1,88 @@
+import argparse
+import re
+import xml.etree.ElementTree as ET
+from pre_commit_hooks.exceptions import PreCommitException
+
+
+def tc_version_pinned(xml_content: str) -> bool:
+    root = ET.fromstring(xml_content)
+
+    return 'TcVersionFixed' in root.attrib and root.attrib.get('TcVersionFixed') == 'true'
+
+
+def get_tc_version(xml_content: str) -> str:
+    root = ET.fromstring(xml_content)
+
+    return root.attrib.get('TcVersion')
+
+
+def fix_tc_version(xml_content: str, new_version: str) -> str:
+    pattern = r'(TcVersion=")([^"]*)(")'
+    new_xml_content = re.sub(pattern, r'\g<1>' + new_version + r'\g<3>', xml_content)
```

Comment:
I like this regex substitution approach because it won't have the same sorts of "incidental" changes that we get when we write out the xml again using lxml.

Reply:
Yeah, I noticed while developing an earlier hook. It was difficult to keep the xmls the same.